src/spanningtrees/parallel_kruskal.jl

```diff
@@ -1,22 +1,111 @@
+The connected componenets are represented in `connected_vs` in the format of disjoint set 
+data-structures.
+Atmost `nthreads` threads will be used.
+The indices thread `i` will iterate over depends upon `local_ind[i]`.
+
+### Implementation Notes
+Thread 'i' will iterate over indices `local_ind[i]`, `local_ind[i]+n_threads`, `local_ind[i]+2*n_threads`....
+Hence, it is required that:
+1. {`rem(local_ind[i], n_threads) for i in 1:n_threads`} = {`i-1 for i in 1:n_threads`}.
+2. The edges at indices `local_ind[i]-n_threads`, `local_ind[i]-2*n_threads` are not cross edges. 
+"""
+function parallel_get_cross_edge(
     edge_list::Vector{Edge{U}},
-    connected_vs::Vector{U},
-    n_threads::Integer
-    current_ind::Vector{R}
+    connected_vs::Vector{Atomic{U}},
+    n_threads::R,
+    local_ind::Vector{R}
     ) where U <: Integer where R <: Integer
 
-    best_ind = Atomic{R}(typemax(R))
+    best_ind = Atomic{R}(length(edge_list)+1)
 
-    @threads for i in 1:n_threads
+    @threads for id in 1:n_threads
+  
+        #Each threads checks an edge with step size = n_threads
+        ind = local_ind[id]
+        while ind < best_ind[]
 
-        id = threadid()
-        for ind in Iterators.countfrom(current_ind[id], n_threads)
+            u = edge_list[ind].src
+            v = edge_list[ind].dst
 
+            #Set connected_vs[u][], connected_vs[v][] to the root of set u, v belong to respectively
+            for i in [u, v]  
+                root = i
+                parent_root = connected_vs[root][]
+                while parent_root != root
+                    root = parent_root
+                    parent_root = connected_vs[root][]
+                end
 
+                p_1 = i
+                p_2 = connected_vs[i][]
+            
+                while p_2 != root
+                    connected_vs[p_1][] = root
+                    p_1 = p_2
+                    p_2 = connected_vs[p_1][]
+                end
+            end
+
+           
+            #If edge_list[ind] is a cross edge
+            if connected_vs[u][] != connected_vs[v][]
+                atomic_min!(best_ind, ind)
+                break
+            end
+            ind += n_threads
         end
+        local_ind[id] = ind
+    end
+
+    return best_ind[]
+end
+
+"""
+    kruskal_mst(g, distmx=weights(g))
+Return a vector of edges representing the minimum spanning tree of a connected, undirected graph `g` with optional
+distance matrix `distmx` using [Kruskal's algorithm](https://en.wikipedia.org/wiki/Kruskal%27s_algorithm).
+"""
+function parallel_kruskal_mst end
+# see https://github.com/mauro3/SimpleTraits.jl/issues/47#issuecomment-327880153 for syntax
+@traitfn function parallel_kruskal_mst(g::AG::(!IsDirected),
+    distmx::AbstractMatrix{T}=weights(g)) where {T <: Real, U, AG <: AbstractGraph{U}}
+
+    nvg = nv(g)
+    neg = ne(g)
+    mst = Vector{Edge}()
+    sizehint!(mst, nvg - 1)
+
+    weights = Vector{T}()
+    sizehint!(weights, neg)
```

Comment:
So something that I noticed about Julia is that your code runs faster if you do
```
v = Vector{MyType}(n)
for i in 1:n
   v[i] = ...
end
```
instead of 
```
v = Vector{MyType}()
sizehint!(v, n)
for i in 1:n
     push!(v, ....)
end
```
I think this is because `push!` does a lot of calculations and checks but I'm not completely sure about that.

Reply:
Yeah that would probably help